core/src/main/java/io/hyperfoil/core/impl/statistics/StatisticsStore.java

```diff
@@ -84,12 +85,309 @@ public void record(String address, int phaseId, int stepId, String metric, Stati
+
+         if(iterValue.getPercentileLevelIteratedTo() != 100.0D){
+            jGenerator.writeNumberField("1/(1-Percentile)",1.0D / (1.0D - iterValue.getPercentileLevelIteratedTo() / 100.0D));
+         }else{
+            jGenerator.writeStringField("1/(1-Percentile)","Infinity");
+         }
+         jGenerator.writeEndObject();
+      }
+      jGenerator.writeEndArray();//end histogram
+   }
+   public void seriesArray(JsonGenerator jGenerator, List<StatisticsSummary> series) throws IOException{
+      jGenerator.writeStartArray();//series
+      if(series!=null){
+         for(StatisticsSummary summary : series){
+            jGenerator.writeStartObject();
+            jGenerator.writeNumberField("Start",summary.startTime);
+            jGenerator.writeNumberField("End",summary.endTime);
+
+            jGenerator.writeNumberField("Requests",summary.requestCount);
+            jGenerator.writeNumberField("Responses",summary.responseCount);
+            jGenerator.writeNumberField("Mean",summary.meanResponseTime);
+            jGenerator.writeNumberField("Min",summary.minResponseTime);
+            Long values[] = summary.percentileResponseTime.values().toArray(new Long[0]);
+            for(int i=0; i<percentiles.length; i++){
+               double p = percentiles[i];
+               double amt = values[i];
+               jGenerator.writeNumberField("p"+(p*100),amt);
+            }
+
+            jGenerator.writeNumberField("Max",summary.minResponseTime);
+            jGenerator.writeNumberField("MeanSendTime",summary.meanSendTime);
+            jGenerator.writeNumberField("ConnFailure",summary.connectFailureCount);
+            jGenerator.writeNumberField("Reset",summary.resetCount);
+            jGenerator.writeNumberField("Timeouts",summary.timeouts);
+            jGenerator.writeNumberField("2xx",summary.status_2xx);
+            jGenerator.writeNumberField("3xx",summary.status_3xx);
+            jGenerator.writeNumberField("4xx",summary.status_4xx);
+            jGenerator.writeNumberField("5xx",summary.status_5xx);
+            jGenerator.writeNumberField("Other",summary.status_other);
+            jGenerator.writeNumberField("Invalid",summary.invalid);
+            jGenerator.writeNumberField("CacheHits",summary.cacheHits);
+            jGenerator.writeNumberField("BlockedCount",summary.blockedCount);
+            jGenerator.writeNumberField("BlockedTime",summary.blockedTime);
+
+            jGenerator.writeEndObject();
+         }
+      }
+      jGenerator.writeEndArray();//end series
+   }
+   public void totalArray(JsonGenerator jGenerator, Data[] dataList,Function<Data,StatisticsSummary> getSummary, BiConsumer<JsonGenerator,Data> also) throws IOException{
+      jGenerator.writeStartArray();
+      for(Data data : dataList){
+         StatisticsSummary summary = getSummary.apply(data);//data.total.summary(percentiles);
+         if(summary==null){
+            continue;
+         }
+
+         jGenerator.writeStartObject();
```

Comment:
I'm leery of directly writing the object because then we expose member field names as public API in the json. Are we comfortable with the member field names being public API?

Reply:
Yes, we already expose `StatisticsSummary` as JSON through REST interface. The object already hosts some JSON annotations.